integration-tests/emulated/tests/assets/asset-hub-kusama/src/tests/treasury.rs

```diff
@@ -0,0 +1,269 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::*;
+use emulated_integration_tests_common::accounts::{ALICE, BOB};
+use frame_support::{
+	dispatch::RawOrigin,
+	sp_runtime::traits::Dispatchable,
+	traits::{
+		fungible::Inspect,
+		fungibles::{Create, Inspect as FungiblesInspect, Mutate},
+	},
+};
+use kusama_runtime::OriginCaller;
+use kusama_runtime_constants::currency::GRAND;
+use runtime_common::impls::VersionedLocatableAsset;
+use xcm_executor::traits::ConvertLocation;
+
+// Fund Treasury account on Asset Hub from Treasury account on Relay Chain with KSMs.
+#[test]
+fn spend_ksm_on_asset_hub() {
+	// initial treasury balance on Asset Hub in KSMs.
+	let treasury_balance = 9_000 * GRAND;
+	// the balance spend on Asset Hub.
+	let treasury_spend_balance = 1_000 * GRAND;
+
+	let init_alice_balance = AssetHubKusama::execute_with(|| {
+		<<AssetHubKusama as AssetHubKusamaPallet>::Balances as Inspect<_>>::balance(
+			&Kusama::account_id_of(ALICE),
+		)
+	});
+
+	Kusama::execute_with(|| {
+		type RuntimeEvent = <Kusama as Chain>::RuntimeEvent;
+		type RuntimeCall = <Kusama as Chain>::RuntimeCall;
+		type Runtime = <Kusama as Chain>::Runtime;
+		type Balances = <Kusama as KusamaPallet>::Balances;
+		type Treasury = <Kusama as KusamaPallet>::Treasury;
+
+		// Fund Treasury account on Asset Hub with KSMs.
+
+		let root = <Kusama as Chain>::RuntimeOrigin::root();
+		let treasury_account = Treasury::account_id();
+
+		// Mist assets to Treasury account on Relay Chain.
+		assert_ok!(Balances::force_set_balance(
+			root.clone(),
+			treasury_account.clone().into(),
+			treasury_balance * 2,
+		));
+
+		let native_asset = Location::here();
+		let asset_hub_location: Location = [Parachain(1000)].into();
+		let treasury_location_on_asset_hub: Location = (Parent, PalletInstance(18)).into();
```

Comment:
The name is confusing. Previous variables are relative to the relay, this one starts with `Parent` but says `on_asset_hub`, leading to believe that it should be relative to the relay as well.

Looks like you want the relay's treasury sovereign account. I'd probably name it `relay_treasury_location` and add a comment.

Reply:
I try to avoid a prefix relay for the treasury. We have only one the Treasury, or the Polkadot/Kusama Treasury, which eventually will move from Relay chain. I rename it to `treasury_location`.